Core.ApplicationServices/SystemUsage/Migration/ItSystemUsageMigrationServiceAdapter.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Core.Abstractions.Types;
+using Core.ApplicationServices.Authorization;
+using Core.ApplicationServices.Model.SystemUsage.Migration;
+using Core.ApplicationServices.Organizations;
+using Core.ApplicationServices.Shared;
+using Core.DomainModel.ItSystem;
+using Core.DomainModel.ItSystemUsage;
+using Core.DomainServices.Generic;
+using Core.DomainServices.Queries;
+using Infrastructure.Services.DataAccess;
+
+namespace Core.ApplicationServices.SystemUsage.Migration
+{
+    public class ItSystemUsageMigrationServiceAdapter : IItSystemUsageMigrationServiceAdapter
+    {
+        private readonly IEntityIdentityResolver _identityResolver;
+        private readonly IItSystemUsageMigrationService _systemUsageMigrationService;
+        private readonly IAuthorizationContext _authorizationContext;
+        private readonly ITransactionManager _transactionManager;
+        private readonly IOrganizationService _organizationService;
+
+        public ItSystemUsageMigrationServiceAdapter(IEntityIdentityResolver identityResolver,
+            IItSystemUsageMigrationService systemUsageMigrationService,
+            IAuthorizationContext authorizationContext,
+            ITransactionManager transactionManager,
+            IOrganizationService organizationService)
+        {
+            _identityResolver = identityResolver;
+            _systemUsageMigrationService = systemUsageMigrationService;
+            _authorizationContext = authorizationContext;
+            _transactionManager = transactionManager;
+            _organizationService = organizationService;
+        }
+
+        public Result<ItSystemUsageMigration, OperationError> GetMigration(Guid usageUuid, Guid toSystemUuid)
+        {
+            return ResolveUsageAndSystemIds(usageUuid, toSystemUuid)
+                .Bind(result => _systemUsageMigrationService.GetSystemUsageMigration(result.usageId, result.systemId))
+                .Bind(migration => WithReadAccess(migration.SystemUsage)
```

Comment:
authorizing After the fact. also this is an adapter, with the purpose of adapting the interface and since you call one function in each just let the application service deal with the responsibility of auth

Reply:
https://refactoring.guru/design-patterns/adapter